# Analyzing Drag Race Success

<a id="table_of_contents"></a>

### Table of contents

<ol>
  <li><a href="#overview">Situation Overview</a>
  <ul>
   <li><a href='#dataset'>Examine the overall dataset</a></li>
    <li><a href='#cleaning'>Execute cleaning procedures</a></li>
  </ul>
  </li>
  
  <li><a href="#eda">Exploratory analysis</a>
  <ul>
    <li><a href='#eda'>How old are the contestants?</a></li>
    <li><a href='#eda'>What is the average age of all the contestants on this show?</a></li>
     <li><a href='#eda'>How many contestants are participating, and where do they come from?</a></li>
     <li><a href='#eda'>What is the averange score?</a></li>
     <li><a href='#eda'>What is the typical number of challenges won on average?</a></li>
    </ul>
  </li>  
  
  <li><a href="#analysis">Data analysis</a>
  <ul>
    <li><a href='#analysis'>Where do the winners come from?</a></li>
    <li><a href='#analysis'>What's the expected number of challenges that the top 4 participants should win, on average, over the course of the competition?</a></li>
    <li><a href='#analysis'>To make it to the top 4, what's the average score a participant should aim for in each episode of the season?</a></li>
    <li><a href='#analysis'>What is the average age of the top 4 participants in the competitions?</a></li>
    <li><a href='#analysis'>Top 4 contestants key elements table </a></li>
    </ul>
  </li>  
  <li><a href="#insights">Insights</a>
</ol>


<a id="overview"></a>
## Situation Overview
The goal of this project is to uncover the key factors, using the available data, that contribute to a participant reaching at least the semifinals in the contest.

<a href="#table_of_contents">Navigate to contents</a>

<a id="dataset"></a>
Examine the overall dataset
<br>
<a href="#table_of_contents">Navigate to contents</a>

In [1]:
#load sql extension
%load_ext sql

In [2]:
#connect to mysql database
%sql mysql://root:***@localhost:3306/project

In [3]:
%%sql 
select * from la_mas_draga 
limit 3

 * mysql://root:***@localhost:3306/project
3 rows affected.


[('1', 'Alexis 3XL', 'Itzel Moreno', 'Matamoros, Tamaulipas', 28, 'Audiciones MTY', 2, 'Ganadora', 2),
 ('2', 'Sophia Jiménez', 'Guillermo Jiménez', 'Guadalajara, Jal.', 31, 'Audiciones GDL', 3, 'Finalistas', 2),
 ('3', 'Gvajardo', 'Pablo Guajardo', 'Monterrey, N.L.', 29, 'Secretísima', 1, 'Finalistas', 2)]

In [4]:
%%sql 
select * from la_mas_draga_scores
limit 3

 * mysql://root:***@localhost:3306/project
3 rows affected.


[('Aisha Dollkills', 1, 'Artesanal', 'MENOS', 7),
 ('Aisha Dollkills', 2, 'Juguete', 'SALV', 14),
 ('Aisha Dollkills', 3, 'A Color', 'ELIM', 12)]

<a id="cleaning"></a>
Execute cleaning procedures
<br>
<a href="#table_of_contents">Navigate to contents</a>

In [38]:
%%sql 
select lugar, participante, temporada
from la_mas_draga
where participante = "Paper cut"

 * mysql://root:***@localhost:3306/project
2 rows affected.


[('12', 'Paper Cut', 4), ('2', 'Paper Cut', 5)]

In [39]:
%%sql 
select Concursante, episodio, progreso, calificacion
from la_mas_draga_scores
where concursante = "Paper cut"

 * mysql://root:***@localhost:3306/project
40 rows affected.


[('Paper Cut', 1, 'ALTO', 17),
 ('Paper Cut', 1, 'ALTO', 0),
 ('Paper Cut', 1, 'ENTRA', 17),
 ('Paper Cut', 1, 'ENTRA', 0),
 ('Paper Cut', 2, 'ALTO', 18),
 ('Paper Cut', 2, 'ALTO', 16),
 ('Paper Cut', 2, 'SALV', 18),
 ('Paper Cut', 2, 'SALV', 16),
 ('Paper Cut', 3, 'EL MÁS', 19),
 ('Paper Cut', 3, 'EL MÁS', 16),
 ('Paper Cut', 3, 'SALV', 19),
 ('Paper Cut', 3, 'SALV', 16),
 ('Paper Cut', 4, 'ALTO', 19),
 ('Paper Cut', 4, 'ALTO', 19),
 ('Paper Cut', 4, 'EL MÁS', 19),
 ('Paper Cut', 4, 'EL MÁS', 19),
 ('Paper Cut', 5, 'ELIM', 8),
 ('Paper Cut', 5, 'ELIM', 18),
 ('Paper Cut', 5, 'ALTO', 8),
 ('Paper Cut', 5, 'ALTO', 18),
 ('Paper Cut', 6, 'FUERA', 0),
 ('Paper Cut', 6, 'FUERA', 20),
 ('Paper Cut', 6, 'EL MÁS', 0),
 ('Paper Cut', 6, 'EL MÁS', 20),
 ('Paper Cut', 7, 'FUERA', 0),
 ('Paper Cut', 7, 'FUERA', 19),
 ('Paper Cut', 7, 'ALTO', 0),
 ('Paper Cut', 7, 'ALTO', 19),
 ('Paper Cut', 8, 'FUERA', 0),
 ('Paper Cut', 8, 'FUERA', 18),
 ('Paper Cut', 8, 'ALTO', 0),
 ('Paper Cut', 8, 'ALTO', 18),
 ('Paper Cut', 9, 'FUERA', 0),
 ('Paper Cut', 9, 'FUERA', 18),
 ('Paper Cut', 9, 'ALTO', 0),
 ('Paper Cut', 9, 'ALTO', 18),
 ('Paper Cut', 10, 'FUERA', 0),
 ('Paper Cut', 10, 'FUERA', 19),
 ('Paper Cut', 10, 'ALTO', 0),
 ('Paper Cut', 10, 'ALTO', 19)]

In [33]:
%%sql 
alter table la_mas_draga_scores
add column unique_id INT AUTO_INCREMENT PRIMARY KEY

 * mysql://root:***@localhost:3306/project
0 rows affected.


[]

In [41]:
%%sql 
alter table la_mas_draga
add column unique_id INT AUTO_INCREMENT PRIMARY KEY

 * mysql://root:***@localhost:3306/project
0 rows affected.


[]

In [40]:
%%sql 
select * from la_mas_draga_scores
limit 3

 * mysql://root:***@localhost:3306/project
3 rows affected.


[('Huma Kyle', 6, 'Del Toro', 'ABDICÓ', 0, 1),
 ('Aisha Dollkills', 1, 'Artesanal', 'MENOS', 7, 2),
 ('Huma Kyle', 7, 'Tejocote', 'FUERA', 0, 3)]

In [42]:
%%sql 
select * from la_mas_draga
limit 3

 * mysql://root:***@localhost:3306/project
3 rows affected.


[('1', 'Alexis 3XL', 'Itzel Moreno', 'Matamoros, Tamaulipas', 28, 'Audiciones MTY', 2, 'Ganadora', 2, 1),
 ('2', 'Sophia Jiménez', 'Guillermo Jiménez', 'Guadalajara, Jal.', 31, 'Audiciones GDL', 3, 'Finalistas', 2, 2),
 ('3', 'Gvajardo', 'Pablo Guajardo', 'Monterrey, N.L.', 29, 'Secretísima', 1, 'Finalistas', 2, 3)]

In [44]:
%%sql 
select sc.unique_id, d.lugar, d.participante, `lugar de residencia`, d.edad, `retos ganados`, d.resultado, d.temporada, sc.episodio, sc.progreso, sc.calificacion 
from la_mas_draga as d
join la_mas_draga_scores as sc
on d.participante = sc.concursante
where participante = "Paper Cut"
order by temporada desc

 * mysql://root:***@localhost:3306/project
80 rows affected.


[(286, '2', 'Paper Cut', 'Ciudad De México', 25, 2, 'Finalista', 5, 1, 'ALTO', 17),
 (306, '2', 'Paper Cut', 'Ciudad De México', 25, 2, 'Finalista', 5, 6, 'FUERA', 0),
 (287, '2', 'Paper Cut', 'Ciudad De México', 25, 2, 'Finalista', 5, 1, 'ALTO', 0),
 (307, '2', 'Paper Cut', 'Ciudad De México', 25, 2, 'Finalista', 5, 6, 'FUERA', 20),
 (288, '2', 'Paper Cut', 'Ciudad De México', 25, 2, 'Finalista', 5, 1, 'ENTRA', 17),
 (308, '2', 'Paper Cut', 'Ciudad De México', 25, 2, 'Finalista', 5, 6, 'EL MÁS', 0),
 (289, '2', 'Paper Cut', 'Ciudad De México', 25, 2, 'Finalista', 5, 1, 'ENTRA', 0),
 (309, '2', 'Paper Cut', 'Ciudad De México', 25, 2, 'Finalista', 5, 6, 'EL MÁS', 20),
 (290, '2', 'Paper Cut', 'Ciudad De México', 25, 2, 'Finalista', 5, 2, 'ALTO', 18),
 (310, '2', 'Paper Cut', 'Ciudad De México', 25, 2, 'Finalista', 5, 7, 'FUERA', 0),
 (291, '2', 'Paper Cut', 'Ciudad De México', 25, 2, 'Finalista', 5, 2, 'ALTO', 16),
 (311, '2', 'Paper Cut', 'Ciudad De México', 25, 2, 'Finalista', 5, 7, 'FUERA', 19),
 (292, '2', 'Paper Cut', 'Ciudad De México', 25, 2, 'Finalista', 5, 2, 'SALV', 18),
 (312, '2', 'Paper Cut', 'Ciudad De México', 25, 2, 'Finalista', 5, 7, 'ALTO', 0),
 (293, '2', 'Paper Cut', 'Ciudad De México', 25, 2, 'Finalista', 5, 2, 'SALV', 16),
 (313, '2', 'Paper Cut', 'Ciudad De México', 25, 2, 'Finalista', 5, 7, 'ALTO', 19),
 (294, '2', 'Paper Cut', 'Ciudad De México', 25, 2, 'Finalista', 5, 3, 'EL MÁS', 19),
 (314, '2', 'Paper Cut', 'Ciudad De México', 25, 2, 'Finalista', 5, 8, 'FUERA', 0),
 (295, '2', 'Paper Cut', 'Ciudad De México', 25, 2, 'Finalista', 5, 3, 'EL MÁS', 16),
 (315, '2', 'Paper Cut', 'Ciudad De México', 25, 2, 'Finalista', 5, 8, 'FUERA', 18),
 (296, '2', 'Paper Cut', 'Ciudad De México', 25, 2, 'Finalista', 5, 3, 'SALV', 19),
 (316, '2', 'Paper Cut', 'Ciudad De México', 25, 2, 'Finalista', 5, 8, 'ALTO', 0),
 (297, '2', 'Paper Cut', 'Ciudad De México', 25, 2, 'Finalista', 5, 3, 'SALV', 16),
 (317, '2', 'Paper Cut', 'Ciudad De México', 25, 2, 'Finalista', 5, 8, 'ALTO', 18),
 (298, '2', 'Paper Cut', 'Ciudad De México', 25, 2, 'Finalista', 5, 4, 'ALTO', 19),
 (318, '2', 'Paper Cut', 'Ciudad De México', 25, 2, 'Finalista', 5, 9, 'FUERA', 0),
 (299, '2', 'Paper Cut', 'Ciudad De México', 25, 2, 'Finalista', 5, 4, 'ALTO', 19),
 (319, '2', 'Paper Cut', 'Ciudad De México', 25, 2, 'Finalista', 5, 9, 'FUERA', 18),
 (300, '2', 'Paper Cut', 'Ciudad De México', 25, 2, 'Finalista', 5, 4, 'EL MÁS', 19),
 (320, '2', 'Paper Cut', 'Ciudad De México', 25, 2, 'Finalista', 5, 9, 'ALTO', 0),
 (301, '2', 'Paper Cut', 'Ciudad De México', 25, 2, 'Finalista', 5, 4, 'EL MÁS', 19),
 (321, '2', 'Paper Cut', 'Ciudad De México', 25, 2, 'Finalista', 5, 9, 'ALTO', 18),
 (302, '2', 'Paper Cut', 'Ciudad De México', 25, 2, 'Finalista', 5, 5, 'ELIM', 8),
 (322, '2', 'Paper Cut', 'Ciudad De México', 25, 2, 'Finalista', 5, 10, 'FUERA', 0),
 (303, '2', 'Paper Cut', 'Ciudad De México', 25, 2, 'Finalista', 5, 5, 'ELIM', 18),
 (323, '2', 'Paper Cut', 'Ciudad De México', 25, 2, 'Finalista', 5, 10, 'FUERA', 19),
 (304, '2', 'Paper Cut', 'Ciudad De México', 25, 2, 'Finalista', 5, 5, 'ALTO', 8),
 (324, '2', 'Paper Cut', 'Ciudad De México', 25, 2, 'Finalista', 5, 10, 'ALTO', 0),
 (305, '2', 'Paper Cut', 'Ciudad De México', 25, 2, 'Finalista', 5, 5, 'ALTO', 18),
 (325, '2', 'Paper Cut', 'Ciudad De México', 25, 2, 'Finalista', 5, 10, 'ALTO', 19),
 (286, '12', 'Paper Cut', 'Ciudad de México', 23, 1, '4° eliminado', 4, 1, 'ALTO', 17),
 (287, '12', 'Paper Cut', 'Ciudad de México', 23, 1, '4° eliminado', 4, 1, 'ALTO', 0),
 (288, '12', 'Paper Cut', 'Ciudad de México', 23, 1, '4° eliminado', 4, 1, 'ENTRA', 17),
 (289, '12', 'Paper Cut', 'Ciudad de México', 23, 1, '4° eliminado', 4, 1, 'ENTRA', 0),
 (290, '12', 'Paper Cut', 'Ciudad de México', 23, 1, '4° eliminado', 4, 2, 'ALTO', 18),
 (291, '12', 'Paper Cut', 'Ciudad de México', 23, 1, '4° eliminado', 4, 2, 'ALTO', 16),
 (292, '12', 'Paper Cut', 'Ciudad de México', 23, 1, '4° eliminado', 4, 2, 'SALV', 18),
 (293, '12

In [46]:
%%sql 
update la_mas_draga_scores
set progreso = "FUERA"
where concursante = "Paper Cut" and calificacion = 0

 * mysql://root:***@localhost:3306/project
12 rows affected.


[]

In [48]:
%%sql 
select sc.unique_id, d.lugar, d.participante, `lugar de residencia`, d.edad, `retos ganados`, d.resultado, d.temporada, sc.episodio, sc.progreso, sc.calificacion 
from la_mas_draga as d
join la_mas_draga_scores as sc
on d.participante = sc.concursante
where d.participante = "Paper Cut" and d.temporada = 5
order by episodio desc

 * mysql://root:***@localhost:3306/project
40 rows affected.


[(325, '2', 'Paper Cut', 'Ciudad De México', 25, 2, 'Finalista', 5, 10, 'ALTO', 19),
 (324, '2', 'Paper Cut', 'Ciudad De México', 25, 2, 'Finalista', 5, 10, 'FUERA', 0),
 (323, '2', 'Paper Cut', 'Ciudad De México', 25, 2, 'Finalista', 5, 10, 'FUERA', 19),
 (322, '2', 'Paper Cut', 'Ciudad De México', 25, 2, 'Finalista', 5, 10, 'FUERA', 0),
 (321, '2', 'Paper Cut', 'Ciudad De México', 25, 2, 'Finalista', 5, 9, 'ALTO', 18),
 (320, '2', 'Paper Cut', 'Ciudad De México', 25, 2, 'Finalista', 5, 9, 'FUERA', 0),
 (319, '2', 'Paper Cut', 'Ciudad De México', 25, 2, 'Finalista', 5, 9, 'FUERA', 18),
 (318, '2', 'Paper Cut', 'Ciudad De México', 25, 2, 'Finalista', 5, 9, 'FUERA', 0),
 (317, '2', 'Paper Cut', 'Ciudad De México', 25, 2, 'Finalista', 5, 8, 'ALTO', 18),
 (316, '2', 'Paper Cut', 'Ciudad De México', 25, 2, 'Finalista', 5, 8, 'FUERA', 0),
 (315, '2', 'Paper Cut', 'Ciudad De México', 25, 2, 'Finalista', 5, 8, 'FUERA', 18),
 (314, '2', 'Paper Cut', 'Ciudad De México', 25, 2, 'Finalista', 5, 8, 'FUERA', 0),
 (313, '2', 'Paper Cut', 'Ciudad De México', 25, 2, 'Finalista', 5, 7, 'ALTO', 19),
 (312, '2', 'Paper Cut', 'Ciudad De México', 25, 2, 'Finalista', 5, 7, 'FUERA', 0),
 (311, '2', 'Paper Cut', 'Ciudad De México', 25, 2, 'Finalista', 5, 7, 'FUERA', 19),
 (310, '2', 'Paper Cut', 'Ciudad De México', 25, 2, 'Finalista', 5, 7, 'FUERA', 0),
 (306, '2', 'Paper Cut', 'Ciudad De México', 25, 2, 'Finalista', 5, 6, 'FUERA', 0),
 (307, '2', 'Paper Cut', 'Ciudad De México', 25, 2, 'Finalista', 5, 6, 'FUERA', 20),
 (308, '2', 'Paper Cut', 'Ciudad De México', 25, 2, 'Finalista', 5, 6, 'FUERA', 0),
 (309, '2', 'Paper Cut', 'Ciudad De México', 25, 2, 'Finalista', 5, 6, 'EL MÁS', 20),
 (305, '2', 'Paper Cut', 'Ciudad De México', 25, 2, 'Finalista', 5, 5, 'ALTO', 18),
 (304, '2', 'Paper Cut', 'Ciudad De México', 25, 2, 'Finalista', 5, 5, 'ALTO', 8),
 (303, '2', 'Paper Cut', 'Ciudad De México', 25, 2, 'Finalista', 5, 5, 'ELIM', 18),
 (302, '2', 'Paper Cut', 'Ciudad De México', 25, 2, 'Finalista', 5, 5, 'ELIM', 8),
 (301, '2', 'Paper Cut', 'Ciudad De México', 25, 2, 'Finalista', 5, 4, 'EL MÁS', 19),
 (300, '2', 'Paper Cut', 'Ciudad De México', 25, 2, 'Finalista', 5, 4, 'EL MÁS', 19),
 (299, '2', 'Paper Cut', 'Ciudad De México', 25, 2, 'Finalista', 5, 4, 'ALTO', 19),
 (298, '2', 'Paper Cut', 'Ciudad De México', 25, 2, 'Finalista', 5, 4, 'ALTO', 19),
 (297, '2', 'Paper Cut', 'Ciudad De México', 25, 2, 'Finalista', 5, 3, 'SALV', 16),
 (296, '2', 'Paper Cut', 'Ciudad De México', 25, 2, 'Finalista', 5, 3, 'SALV', 19),
 (295, '2', 'Paper Cut', 'Ciudad De México', 25, 2, 'Finalista', 5, 3, 'EL MÁS', 16),
 (294, '2', 'Paper Cut', 'Ciudad De México', 25, 2, 'Finalista', 5, 3, 'EL MÁS', 19),
 (290, '2', 'Paper Cut', 'Ciudad De México', 25, 2, 'Finalista', 5, 2, 'ALTO', 18),
 (293, '2', 'Paper Cut', 'Ciudad De México', 25, 2, 'Finalista', 5, 2, 'SALV', 16),
 (292, '2', 'Paper Cut', 'Ciudad De México', 25, 2, 'Finalista', 5, 2, 'SALV', 18),
 (291, '2', 'Paper Cut', 'Ciudad De México', 25, 2, 'Finalista', 5, 2, 'ALTO', 16),
 (287, '2', 'Paper Cut', 'Ciudad De México', 25, 2, 'Finalista', 5, 1, 'FUERA', 0),
 (288, '2', 'Paper Cut', 'Ciudad De México', 25, 2, 'Finalista', 5, 1, 'ENTRA', 17),
 (289, '2', 'Paper Cut', 'Ciudad De México', 25, 2, 'Finalista', 5, 1, 'FUERA', 0),
 (286, '2', 'Paper Cut', 'Ciudad De México', 25, 2, 'Finalista', 5, 1, 'ALTO', 17)]

In [49]:
%%sql 
delete from la_mas_draga_scores
where unique_id in ("286", "288", "287","291", "292", "290", "294", "295", "296", "298", "299", "301", "302", "303", "304", "308", "307", "306", "310", "311", "312", "314", "315", "316", "318", "319", "320", "322", "323", "324" )

 * mysql://root:***@localhost:3306/project
30 rows affected.


[]

In [50]:
%%sql 
select sc.unique_id, d.lugar, d.participante, `lugar de residencia`, d.edad, `retos ganados`, d.resultado, d.temporada, sc.episodio, sc.progreso, sc.calificacion 
from la_mas_draga as d
join la_mas_draga_scores as sc
on d.participante = sc.concursante
where d.participante = "Paper Cut" and d.temporada = 5
order by episodio desc

 * mysql://root:***@localhost:3306/project
10 rows affected.


[(325, '2', 'Paper Cut', 'Ciudad De México', 25, 2, 'Finalista', 5, 10, 'ALTO', 19),
 (321, '2', 'Paper Cut', 'Ciudad De México', 25, 2, 'Finalista', 5, 9, 'ALTO', 18),
 (317, '2', 'Paper Cut', 'Ciudad De México', 25, 2, 'Finalista', 5, 8, 'ALTO', 18),
 (313, '2', 'Paper Cut', 'Ciudad De México', 25, 2, 'Finalista', 5, 7, 'ALTO', 19),
 (309, '2', 'Paper Cut', 'Ciudad De México', 25, 2, 'Finalista', 5, 6, 'EL MÁS', 20),
 (305, '2', 'Paper Cut', 'Ciudad De México', 25, 2, 'Finalista', 5, 5, 'ALTO', 18),
 (300, '2', 'Paper Cut', 'Ciudad De México', 25, 2, 'Finalista', 5, 4, 'EL MÁS', 19),
 (297, '2', 'Paper Cut', 'Ciudad De México', 25, 2, 'Finalista', 5, 3, 'SALV', 16),
 (293, '2', 'Paper Cut', 'Ciudad De México', 25, 2, 'Finalista', 5, 2, 'SALV', 16),
 (289, '2', 'Paper Cut', 'Ciudad De México', 25, 2, 'Finalista', 5, 1, 'FUERA', 0)]

In [51]:
%%sql 
select sc.unique_id, d.lugar, d.participante, `lugar de residencia`, d.edad, `retos ganados`, d.resultado, d.temporada, sc.episodio, sc.progreso, sc.calificacion 
from la_mas_draga as d
join la_mas_draga_scores as sc
on d.participante = sc.concursante
where d.participante = "Paper Cut" and d.temporada = 4
order by episodio desc

 * mysql://root:***@localhost:3306/project
10 rows affected.


[(325, '12', 'Paper Cut', 'Ciudad de México', 23, 1, '4° eliminado', 4, 10, 'ALTO', 19),
 (321, '12', 'Paper Cut', 'Ciudad de México', 23, 1, '4° eliminado', 4, 9, 'ALTO', 18),
 (317, '12', 'Paper Cut', 'Ciudad de México', 23, 1, '4° eliminado', 4, 8, 'ALTO', 18),
 (313, '12', 'Paper Cut', 'Ciudad de México', 23, 1, '4° eliminado', 4, 7, 'ALTO', 19),
 (309, '12', 'Paper Cut', 'Ciudad de México', 23, 1, '4° eliminado', 4, 6, 'EL MÁS', 20),
 (305, '12', 'Paper Cut', 'Ciudad de México', 23, 1, '4° eliminado', 4, 5, 'ALTO', 18),
 (300, '12', 'Paper Cut', 'Ciudad de México', 23, 1, '4° eliminado', 4, 4, 'EL MÁS', 19),
 (297, '12', 'Paper Cut', 'Ciudad de México', 23, 1, '4° eliminado', 4, 3, 'SALV', 16),
 (293, '12', 'Paper Cut', 'Ciudad de México', 23, 1, '4° eliminado', 4, 2, 'SALV', 16),
 (289, '12', 'Paper Cut', 'Ciudad de México', 23, 1, '4° eliminado', 4, 1, 'FUERA', 0)]

In [55]:
%%sql 
update la_mas_draga_scores
set progreso = "FUERA", calificacion = 0
where concursante = "Paper Cut" and unique_id in ("309", "313", "317", "321", "325")


 * mysql://root:***@localhost:3306/project
5 rows affected.


[]

In [56]:
%%sql 
select sc.unique_id, d.lugar, d.participante, `lugar de residencia`, d.edad, `retos ganados`, d.resultado, d.temporada, sc.episodio, sc.progreso, sc.calificacion 
from la_mas_draga as d
join la_mas_draga_scores as sc
on d.participante = sc.concursante
where d.participante = "Paper Cut" and d.temporada = 4
order by episodio desc

 * mysql://root:***@localhost:3306/project
10 rows affected.


[(325, '12', 'Paper Cut', 'Ciudad de México', 23, 1, '4° eliminado', 4, 10, 'FUERA', 0),
 (321, '12', 'Paper Cut', 'Ciudad de México', 23, 1, '4° eliminado', 4, 9, 'FUERA', 0),
 (317, '12', 'Paper Cut', 'Ciudad de México', 23, 1, '4° eliminado', 4, 8, 'FUERA', 0),
 (313, '12', 'Paper Cut', 'Ciudad de México', 23, 1, '4° eliminado', 4, 7, 'FUERA', 0),
 (309, '12', 'Paper Cut', 'Ciudad de México', 23, 1, '4° eliminado', 4, 6, 'FUERA', 0),
 (305, '12', 'Paper Cut', 'Ciudad de México', 23, 1, '4° eliminado', 4, 5, 'ALTO', 18),
 (300, '12', 'Paper Cut', 'Ciudad de México', 23, 1, '4° eliminado', 4, 4, 'EL MÁS', 19),
 (297, '12', 'Paper Cut', 'Ciudad de México', 23, 1, '4° eliminado', 4, 3, 'SALV', 16),
 (293, '12', 'Paper Cut', 'Ciudad de México', 23, 1, '4° eliminado', 4, 2, 'SALV', 16),
 (289, '12', 'Paper Cut', 'Ciudad de México', 23, 1, '4° eliminado', 4, 1, 'FUERA', 0)]

In [64]:
%%sql 
select distinct(resultado) from la_mas_draga 

 * mysql://root:***@localhost:3306/project
18 rows affected.


[('Ganadora',),
 ('Finalistas',),
 ('6ª eliminada',),
 ('5ª eliminada',),
 ('4ª eliminada',),
 ('3ª eliminadas',),
 ('2ª eliminada',),
 ('1ª eliminada',),
 ('8ª/9ª eliminada',),
 ('7ª eliminada',),
 ('3ª eliminada',),
 ('Finalista',),
 ('9ª/10ª eliminada',),
 ('8ª eliminada',),
 ('6ª/7ª eliminada',),
 ('4° eliminado',),
 ('Finalista secreta',),
 ('Abandona',)]

In [65]:
%%sql 
select distinct(lugar) from la_mas_draga

 * mysql://root:***@localhost:3306/project
14 rows affected.


[('1',),
 ('2',),
 ('3',),
 ('4',),
 ('5',),
 ('6',),
 ('7',),
 ('8',),
 ('9',),
 ('10',),
 ('11',),
 ('12',),
 ('13',),
 ('14',)]

In [66]:
%%sql
select distinct(progreso) from la_mas_draga_scores

 * mysql://root:***@localhost:3306/project
24 rows affected.


[('ABDICÓ',),
 ('MENOS',),
 ('FUERA',),
 ('SALV',),
 ('ELIM',),
 ('VUELVE',),
 ('ALTA',),
 ('LA MÁS',),
 ('Invitada',),
 ('Compitiendo',),
 ('Ganadora',),
 ('ENTRA',),
 ('BAJAS',),
 ('ALTAS',),
 ('Invitadas',),
 ('BAJA',),
 ('GANA',),
 ('Finalista',),
 ('ABAND',),
 ('ALTO',),
 ('EL MÁS',),
 ('BAJO',),
 ('Invitado',),
 ('LAS MÁS',)]

In [67]:
%%sql 
update la_mas_draga_scores
set progreso = case
    when progreso = "ALTA MENOS" then "MENOS"
    when progreso = "ALTA EG" then "ALTA"
    when progreso = "SALV MENOS" then "MENOS"
    When progreso = "MENOS SALV" then "SALV"
    When progreso = "MENOS BAJA" then "BAJA"
    else progreso
end 

 * mysql://root:***@localhost:3306/project
433 rows affected.


[]

<a id="eda"></a>
## Exploratory Analysis
To gain understanding of my data, the first step is exploration

<a href="#table_of_contents">Navigate to contents</a>

#### How old are the constestants?

In [68]:
%%sql 
select distinct(edad) from la_mas_draga 

 * mysql://root:***@localhost:3306/project
20 rows affected.


[(28,),
 (31,),
 (29,),
 (34,),
 (33,),
 (25,),
 (37,),
 (35,),
 (157,),
 (39,),
 (24,),
 (30,),
 (36,),
 (26,),
 (20,),
 (23,),
 (43,),
 (32,),
 (27,),
 (21,)]

#### What is the average age of all the contestants on this show?

In [69]:
%%sql
select round(avg(edad)) as avg_edad 
from la_mas_draga

 * mysql://root:***@localhost:3306/project
1 rows affected.


[(Decimal('32'),)]

#### How many contestants are participating, and where do they come from?

In [70]:
%%sql
select `lugar de residencia` as place, COUNT(*) as participant_count
from la_mas_draga
group by `lugar de residencia`
order by participant_count desc

 * mysql://root:***@localhost:3306/project
23 rows affected.


[('Ciudad de México', 11),
 ('Monterrey, N.L.', 9),
 ('Guadalajara, Jal.', 7),
 ('Chihuahua, Chih.', 2),
 ('Acapulco de Juárez, Gro.', 2),
 ('Matamoros, Tamaulipas', 1),
 ('Saltillo, Coah.', 1),
 ('Progreso, Yuc.', 1),
 ('Mérida, Yuc.', 1),
 ('Aguascalientes, Ags.', 1),
 ('Ciudad Juárez, Chih.', 1),
 ('Gómez Palacio, Dgo.', 1),
 ('Antofagasta, Chile', 1),
 ('Ensenada, B.C.', 1),
 ('Mexicali, B.C.', 1),
 ('Zacatecas, Zac.', 1),
 ('Lázaro Cárdenas, Mich.', 1),
 ('Santiago, Chile', 1),
 ('Ciudad Neza, Edo de Méx.', 1),
 ('Medellín, Colombia', 1),
 ('San José, Costa Rica', 1),
 ('Cali, Colombia', 1),
 ('San Diego, California', 1)]

#### What is the averange score?

In [71]:
%%sql 
select round(avg(calificacion)) as avg_score
from la_mas_draga_scores

 * mysql://root:***@localhost:3306/project
1 rows affected.


[(Decimal('9'),)]

#### What is the typical number of challenges won on average?

In [72]:
%%sql 
select round(avg(`retos ganados`),2) as avg_challenges_won
from la_mas_draga

 * mysql://root:***@localhost:3306/project
1 rows affected.


[(Decimal('0.82'),)]

<a id="analysis"></a>
## Data Analysis

<a href="#table_of_contents">Navigate to contents</a>

#### Where do the winners come from?

In [73]:
%%sql 
select `lugar de residencia` as place, count(*) as participant_count
from la_mas_draga
where resultado in ("Ganadora", "Finalista", "Finalistas", "Finalista secreta")
group by `lugar de residencia`
order by participant_count desc

 * mysql://root:***@localhost:3306/project
11 rows affected.


[('Guadalajara, Jal.', 3),
 ('Monterrey, N.L.', 3),
 ('Matamoros, Tamaulipas', 1),
 ('Mérida, Yuc.', 1),
 ('Antofagasta, Chile', 1),
 ('Chihuahua, Chih.', 1),
 ('Ensenada, B.C.', 1),
 ('Lázaro Cárdenas, Mich.', 1),
 ('Ciudad De México', 1),
 ('Acapulco de Juárez, Gro.', 1),
 ('Santiago, Chile', 1)]

#### What's the expected number of challenges that the top 4 participants should win, on average, over the course of the competition?

In [74]:
%%sql 
select round(avg(`retos ganados`)) as wins
from la_mas_draga
where resultado in ("Ganadora", "Finalista", "Finalistas", "Finalista secreta")

 * mysql://root:***@localhost:3306/project
1 rows affected.


[(Decimal('2'),)]

#### To make it to the top 4, what's the average score a participant should aim for in each episode of the season?

In [75]:
%%sql 
select round(avg(s.calificacion),2)
from la_mas_draga_scores as s
join la_mas_draga as d 
on d.participante = s.concursante
where d.resultado in ("Ganadora", "Finalista", "Finalistas", "Finalista secreta")

 * mysql://root:***@localhost:3306/project
1 rows affected.


[(Decimal('13.38'),)]

#### What is the average age of the top 4 participants in the competitions?

In [76]:
%%sql 
select round(avg(edad))
from la_mas_draga
where resultado in ("Ganadora", "Finalista", "Finalistas", "Finalista secreta")

 * mysql://root:***@localhost:3306/project
1 rows affected.


[(Decimal('29'),)]

#### Top 4 contestants key elements table 

In [63]:
%%sql 
select d.lugar, d.participante, `lugar de residencia`, d.edad, `retos ganados`, d.resultado, d.temporada, sc.episodio, sc.progreso, sc.calificacion 
from la_mas_draga as d
join la_mas_draga_scores as sc
on d.participante = sc.concursante
where resultado in ("Ganadora", "Finalista", "Finalistas", "Finalista secreta")
order by temporada asc

 * mysql://root:***@localhost:3306/project
128 rows affected.


[('1', 'Alexis 3XL', 'Matamoros, Tamaulipas', 28, 2, 'Ganadora', 2, 1, 'MENOS', 14),
 ('1', 'Alexis 3XL', 'Matamoros, Tamaulipas', 28, 2, 'Ganadora', 2, 2, 'ALTA', 13),
 ('1', 'Alexis 3XL', 'Matamoros, Tamaulipas', 28, 2, 'Ganadora', 2, 3, 'ALTA', 18),
 ('1', 'Alexis 3XL', 'Matamoros, Tamaulipas', 28, 2, 'Ganadora', 2, 4, 'LA MÁS', 18),
 ('1', 'Alexis 3XL', 'Matamoros, Tamaulipas', 28, 2, 'Ganadora', 2, 5, 'SALV', 15),
 ('1', 'Alexis 3XL', 'Matamoros, Tamaulipas', 28, 2, 'Ganadora', 2, 6, 'LA MÁS', 19),
 ('1', 'Alexis 3XL', 'Matamoros, Tamaulipas', 28, 2, 'Ganadora', 2, 7, 'MENOS', 18),
 ('1', 'Alexis 3XL', 'Matamoros, Tamaulipas', 28, 2, 'Ganadora', 2, 8, 'Compitiendo', 0),
 ('1', 'Alexis 3XL', 'Matamoros, Tamaulipas', 28, 2, 'Ganadora', 2, 9, 'Ganadora', 0),
 ('3', 'Gvajardo', 'Monterrey, N.L.', 29, 1, 'Finalistas', 2, 1, 'LA MÁS', 19),
 ('3', 'Gvajardo', 'Monterrey, N.L.', 29, 1, 'Finalistas', 2, 2, 'MENOS', 8),
 ('3', 'Gvajardo', 'Monterrey, N.L.', 29, 1, 'Finalistas', 2, 3, 'BAJA', 8),
 ('3', 'Gvajardo', 'Monterrey, N.L.', 29, 1, 'Finalistas', 2, 4, 'MENOS', 10),
 ('3', 'Gvajardo', 'Monterrey, N.L.', 29, 1, 'Finalistas', 2, 5, 'SALV', 12),
 ('3', 'Gvajardo', 'Monterrey, N.L.', 29, 1, 'Finalistas', 2, 6, 'ALTA', 16),
 ('3', 'Gvajardo', 'Monterrey, N.L.', 29, 1, 'Finalistas', 2, 7, 'MENOS', 12),
 ('3', 'Gvajardo', 'Monterrey, N.L.', 29, 1, 'Finalistas', 2, 8, 'Compitiendo', 0),
 ('3', 'Gvajardo', 'Monterrey, N.L.', 29, 1, 'Finalistas', 2, 9, 'Finalista', 0),
 ('1', 'Aviesc Who?', 'Guadalajara, Jal.', 37, 1, 'Ganadora', 3, 1, 'ALTA', 14),
 ('1', 'Aviesc Who?', 'Guadalajara, Jal.', 37, 1, 'Ganadora', 3, 2, 'MENOS', 5),
 ('1', 'Aviesc Who?', 'Guadalajara, Jal.', 37, 1, 'Ganadora', 3, 3, 'LA MÁS', 20),
 ('1', 'Aviesc Who?', 'Guadalajara, Jal.', 37, 1, 'Ganadora', 3, 4, 'SALV', 11),
 ('1', 'Aviesc Who?', 'Guadalajara, Jal.', 37, 1, 'Ganadora', 3, 5, 'ELIM', 6),
 ('1', 'Aviesc Who?', 'Guadalajara, Jal.', 37, 1, 'Ganadora', 3, 6, 'FUERA', 0),
 ('1', 'Aviesc Who?', 'Guadalajara, Jal.', 37, 1, 'Ganadora', 3, 7, 'FUERA', 18),
 ('1', 'Aviesc Who?', 'Guadalajara, Jal.', 37, 1, 'Ganadora', 3, 8, 'ALTA', 10),
 ('1', 'Aviesc Who?', 'Guadalajara, Jal.', 37, 1, 'Ganadora', 3, 9, 'MENOS', 0),
 ('1', 'Aviesc Who?', 'Guadalajara, Jal.', 37, 1, 'Ganadora', 3, 10, 'Compitiendo', 0),
 ('3', 'Raga Diamante', 'Mérida, Yuc.', 35, 3, 'Finalistas', 3, 1, 'SALV', 12),
 ('3', 'Raga Diamante', 'Mérida, Yuc.', 35, 3, 'Finalistas', 3, 2, 'LA MÁS', 19),
 ('3', 'Raga Diamante', 'Mérida, Yuc.', 35, 3, 'Finalistas', 3, 3, 'SALV', 16),
 ('3', 'Raga Diamante', 'Mérida, Yuc.', 35, 3, 'Finalistas', 3, 4, 'LA MÁS', 23),
 ('3', 'Raga Diamante', 'Mérida, Yuc.', 35, 3, 'Finalistas', 3, 5, 'ALTA', 17),
 ('3', 'Raga Diamante', 'Mérida, Yuc.', 35, 3, 'Finalistas', 3, 6, 'SALV', 14),
 ('3', 'Raga Diamante', 'Mérida, Yuc.', 35, 3, 'Finalistas', 3, 7, 'LA MÁS', 20),
 ('3', 'Raga Diamante', 'Mérida, Yuc.', 35, 3, 'Finalistas', 3, 8, 'SALV', 15),
 ('3', 'Raga Diamante', 'Mérida, Yuc.', 35, 3, 'Finalistas', 3, 9, 'ALTA', 19),
 ('3', 'Raga Diamante', 'Mérida, Yuc.', 35, 3, 'Finalistas', 3, 10, 'Compitiendo', 0),
 ('4', 'Rudy Reyes', 'Monterrey, N.L.', 28, 1, 'Finalistas', 3, 1, 'SALV', 12),
 ('4', 'Rudy Reyes', 'Monterrey, N.L.', 28, 1, 'Finalistas', 3, 2, 'SALV', 10),
 ('4', 'Rudy Reyes', 'Monterrey, N.L.', 28, 1, 'Finalistas', 3, 3, 'BAJA', 9),
 ('4', 'Rudy Reyes', 'Monterrey, N.L.', 28, 1, 'Finalistas', 3, 4, 'SALV', 15),
 ('4', 'Rudy Reyes', 'Monterrey, N.L.', 28, 1, 'Finalistas', 3, 5, 'BAJA', 10),
 ('4', 'Rudy Reyes', 'Monterrey, N.L.', 28, 1, 'Finalistas', 3, 6, 'MENOS', 7),
 ('4', 'Rudy Reyes', 'Monterrey, N.L.', 28, 1, 'Finalistas', 3, 7, 'ALTA', 19),
 ('4', 'Rudy Reyes', 'Monterrey, N.L.', 28, 1, 'Finalistas', 3, 8, 'SALV', 13),
 ('4', 'Rudy Reyes', 'Monterrey, N.L.', 28, 1, 'Finalistas', 3, 9, 'LA MÁS', 20),
 ('4', 'Rudy Reyes', 'Monterrey, N.L.', 28, 1, 'Finalistas', 3, 10, 'Compitiendo', 0),
 ('3', 'Iris XC', 'Chihuahua, Chih.', 26, 0, 'Finalista', 4, 1, 'SALV', 15

<a id="insights"></a>
## Insights

* Participants should aim for at least two wins during all the competition
* To remain competitive, it's important to maintain an average score of at least 13.38
* The top four contestants predominantly come Guadalajara and Monterrey
* The analysis found that the typical age of the top four contestants averages out to be 29 years old.

<a href="#table_of_contents">Navigate to contents</a>